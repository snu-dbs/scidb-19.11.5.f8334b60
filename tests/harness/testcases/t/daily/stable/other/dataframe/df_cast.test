# Test cast() with dataframe input.

--setup
create temp array DF <v:int32, w:char>
store( \
  project( \
    apply( \
      flatten(build(<v:int32>[i=0:19], i)), \
      (w, char(42))), \
    i, inverse:1), \
  DF)

--test
--set-format tsv
cast(DF, <v:int64, doubleyou:char>)

cast(DF, <v:int64, doubleyou:char>[foo; bar])
--error --code=scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_RESERVED_NAME \
    "cast(build(<v:int32>[i=0:9], i), <dummy:char>[\$seq])"
--error --code=scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_RESERVED_NAME \
    "cast(build(<v:int32>[i=0:9], i), (i, \$seq))"

--cleanup
remove(DF)
