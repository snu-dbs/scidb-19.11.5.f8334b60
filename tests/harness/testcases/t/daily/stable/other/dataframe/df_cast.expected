Query was executed successfully

Query was executed successfully

0	*
1	*
2	*
3	*
4	*
5	*
6	*
7	*
8	*
9	*
10	*
11	*
12	*
13	*
14	*
15	*
16	*
17	*
18	*
19	*

0	*
1	*
2	*
3	*
4	*
5	*
6	*
7	*
8	*
9	*
10	*
11	*
12	*
13	*
14	*
15	*
16	*
17	*
18	*
19	*

[An error expected at this place for the query "cast(build(<v:int32>[i=0:9], i), <dummy:char>[$seq])". And it failed with error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_RESERVED_NAME. Expected error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_RESERVED_NAME.]

[An error expected at this place for the query "cast(build(<v:int32>[i=0:9], i), (i, $seq))". And it failed with error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_RESERVED_NAME. Expected error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_RESERVED_NAME.]

Query was executed successfully

